src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.cs

```diff
@@ -256,5 +257,24 @@ public static void SetDllImportResolver(Assembly assembly, DllImportResolver res
 
             return resolver(libraryName, assembly, hasDllImportSearchPathFlags ? (DllImportSearchPath?)dllImportSearchPathFlags : null);
         }
+
+        [DllImport(RuntimeHelpers.QCall, SetLastError = true)]
+        private static extern IntPtr GetEntryPointModuleHandleInternal();
+
```

Comment:
The icalls are usually put into the vm specific files.

Reply:
One of the things I wanted to do with this PR was start a process of sharing QCalls between the runtimes as per #57369, Mono may be moving to more QCalls in the future. If CoreCLR and Mono are going to have the same signatures with the same behavior, we might as well share the managed implementation even if we can't share the native.

Reply:
> #57369

I think this is wrong issue number.

FWIW, we have tried to use shared QCalls a few times in the past, but it always fired back for various reasons and got undone so far. We can give it another try though.

The solution that is proven to work well is to the cleanup the unmanaged dependencies and move them to libraries/src/native .

Reply:
Yes that is. I meant #57639, sorry.

I'll look at moving the dependencies to the libraries native sources.

Reply:
IIRC the last time QCalls were shared was for Globalization, which was undone due to the changes needed for single file providing an alternative solution.

Reply:
There was also a brief period where QCalls used for tracing were shared.